DependencyProperty/src/main/java/com/wada811/dependencyproperty/DependencyModules.kt

```diff
@@ -18,7 +18,7 @@ class DependencyModules internal constructor(vararg modules: DependencyModule) {
 
     @Suppress("UNCHECKED_CAST")
     internal fun <T : DependencyModule> findModule(clazz: Class<T>): T {
-        return modules.filter { clazz.isAssignableFrom(it.key) }.map { it.value }.firstOrNull() as? T
+        return modules.asSequence().filter { clazz.isAssignableFrom(it.key) }.map { it.value }.firstOrNull() as? T
```

Comment:
As you know, Sequence is better performance in _intermediate_ operation like `filter`.
But, `asSequence()` has some overhead in smaller collection.

ref. https://kotlinlang.org/docs/reference/sequences.html
> However, the lazy nature of sequences adds some overhead which may be significant when processing smaller collections or doing simpler computations.

In this case, I used `firstOrNull()` as _terminal_ operation.
So, we can use `firstOrNull(predicate: (T) -> Boolean)` like below.

```suggestion
        return modules.entries.firstOrNull { clazz.isAssignableFrom(it.key) }?.value as? T
```

I think this suggestion is a little better performance.
What do you think?

Reply:
oh yea I didn't go into the size of `modules` actually and that's cooler 👍 
Let me close the PR then 👍